packages/fuels-core/src/json_abi.rs

```diff
@@ -1299,13 +1379,139 @@ mod tests {
         let encoded = abi
             .encode_with_function_selector(json_abi, function_name, &values)
             .unwrap();
-        println!("encoded: {:?}\n", encoded);
 
         let expected_encode =
             "00000000f40ff3b5000000000000000100000000000000010000000000000002000000000000000a";
         assert_eq!(encoded, expected_encode);
     }
 
+    #[test]
+    fn tuple_encode_and_decode() {
+        let json_abi = r#"
+        [
+            {
+                "type":"contract",
+                "inputs": [
+                  {
+                    "name": "input",
+                    "type": "(u64, u64)",
+                    "components": [
+                      {
+                        "name": "__tuple_element",
+                        "type": "u64",
+                        "components": null
+                      },
+                      {
+                        "name": "__tuple_element",
+                        "type": "u64",
```

Comment:
try with two different types and values as a sanity check

Reply:
Done :)